pkg/authorization/registry/util/impersonate.go

```diff
@@ -0,0 +1,19 @@
+package util
+
+import (
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/client-go/rest"
+	rbacinternalversion "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset/typed/rbac/internalversion"
+
+	authclient "github.com/openshift/origin/pkg/auth/client"
+)
+
+func NewImpersonatingRBACFromContext(ctx apirequest.Context, restclient rest.Interface) (rbacinternalversion.RbacInterface, error) {
```

Comment:
> this is RBAC specific, no much point in making it "generic".
> The generic part is "NewImpersonatingRESTClient"

This is generic to anyone trying to create an impersonating client of this type.  I know there are other spots that want impersonating clients.  It may end up as creating a more complete clientset, but I'd like to gather that structure now rather than having a dozen of these.